platform/ios/src/MGLMapView.mm

```diff
@@ -1459,10 +1459,12 @@ - (void)handlePinchGesture:(UIPinchGestureRecognizer *)pinch
         double zoom = log2(newScale);
         MGLMapCamera *toCamera = [self cameraByZoomingToZoomLevel:zoom aroundAnchorPoint:centerPoint];
 
-        if ([self _shouldChangeFromCamera:oldCamera toCamera:toCamera])
+        if ( ! [self _shouldChangeFromCamera:oldCamera toCamera:toCamera])
```

Comment:
This is the key line — if the response from the delegate is `NO` we shouldn’t drift. Previously, this would always allow drifting because, if the delegate responded `NO` it would move on to the following `else`.